packages/browser-wallet/src/popup/pages/SignCIS3Message/SignCIS3Message.tsx

```diff
@@ -0,0 +1,169 @@
+    signMessage,
+} from '@concordium/web-sdk';
+import { fullscreenPromptContext } from '@popup/page-layouts/FullscreenPromptLayout';
+import { usePrivateKey } from '@popup/shared/utils/account-helpers';
+import { displayUrl } from '@popup/shared/utils/string-helpers';
+import { TextArea } from '@popup/shared/Form/TextArea';
+import ConnectedBox from '@popup/pages/Account/ConnectedBox';
+import Button from '@popup/shared/Button';
+import { addToastAtom } from '@popup/state';
+import ExternalRequestLayout from '@popup/page-layouts/ExternalRequestLayout';
+import { SignMessageObject } from '@concordium/browser-wallet-api-helpers';
+
+const SERIALIZATION_HELPER_SCHEMA =
+    'FAAFAAAAEAAAAGNvbnRyYWN0X2FkZHJlc3MMBQAAAG5vbmNlBQkAAAB0aW1lc3RhbXANCwAAAGVudHJ5X3BvaW50FgEHAAAAcGF5bG9hZBABAg==';
+
+type Props = {
+    onSubmit(signature: AccountTransactionSignature): void;
+    onReject(): void;
+};
+
+interface Location {
+    state: {
+        payload: {
+            accountAddress: string;
+            message: SignMessageObject;
+            url: string;
+            cis3ContractDetails: Cis3ContractDetailsObject;
+        };
+    };
+}
+
+type Cis3ContractDetailsObject = {
+    contractAddress: ContractAddress.Type;
+    contractName: ContractName.Type;
+    entrypointName: EntrypointName.Type;
+    nonce: bigint;
+    expiryTimeSignature: string;
+};
+
+async function parseMessage(message: SignMessageObject) {
+    return stringify(
+        deserializeTypeValue(Buffer.from(message.data, 'hex'), Buffer.from(message.schema, 'base64')),
+        undefined,
+        2
+    );
+}
+
+function serializeMessage(payloadMessage: SignMessageObject, cis3ContractDetails: Cis3ContractDetailsObject) {
+    const { contractAddress, entrypointName, nonce, expiryTimeSignature } = cis3ContractDetails;
+    const message = {
+        contract_address: {
+            index: Number(contractAddress.index),
+            subindex: Number(contractAddress.subindex),
+        },
+        nonce: Number(nonce),
+        timestamp: expiryTimeSignature,
+        entry_point: EntrypointName.toString(entrypointName),
+        payload: Array.from(Buffer.from(payloadMessage.data, 'hex')),
+    };
+
+    return serializeTypeValue(message, Buffer.from(SERIALIZATION_HELPER_SCHEMA, 'base64'));
+}
+
+function MessageDetailsDisplay({
+    payloadMessage,
+    cis3ContractDetails,
+}: {
+    payloadMessage: SignMessageObject;
+    cis3ContractDetails: Cis3ContractDetailsObject;
+}) {
+    const { t } = useTranslation('signCIS3Message');
+    const { contractAddress, contractName, entrypointName, nonce, expiryTimeSignature } = cis3ContractDetails;
+    const [parsedMessage, setParsedMessage] = useState<string>('');
+    const expiry = new Date(expiryTimeSignature).toString();
+
+    useEffect(() => {
+        try {
+            parseMessage(payloadMessage).then((m) => setParsedMessage(m));
+        } catch (e) {
+            setParsedMessage(t('unableToDeserialize'));
+        }
```

Comment:
**Suggestion:** Use `.catch` here.

To my understanding this error handling is unreachable even on error, since `parseMessage` is defined using `async` it will reject the promise instead, and since we don't use `await` in this function, this rejection is never turned into an exception.
I think the simplest solution is to use `.catch` instead of `try-catch` here.
```suggestion
        parseMessage(payloadMessage)
          .then((m) => setParsedMessage(m))
          .catch((_) => setParsedMessage(t('unableToDeserialize')));
```